scripts/convert_ima_policy.py

```diff
@@ -0,0 +1,173 @@
+import argparse
+import copy
+import datetime
+import json
+import re
+import sys
+from os.path import basename
+
+from keylime.ima import file_signatures, ima
+
+"""
+This script converts legacy Keylime policies into the unified, next-generation
+IMA policy format.Both legacy allow/exclude lists as well as out-of-date IMA
+policy formats are accepted.
+
+Example usage:
+
+To convert an allowlist and exclude list to an IMA policy at `keylime-policy.json`:
+
+```
+python convert_ima_policy.py -a <allowlist path> -e <exclude list path> -o keylime-policy.json
+```
+
+To convert an existing IMA policy to the latest version:
+
+```
+python convert_ima_policy.py -i <ima policy path> -o keylime-policy.json
+```
+
+To view help and see all available options:
+
+```
+python convert_ima_policy.py -h
+```
+"""
+
+# Creates an IMA policy from provided legacy allowlist.
+def create_ima_policy(allowlist_path):
+    with open(allowlist_path, "r") as f:
+        alist_raw = f.read()
+
+    p = re.compile(r"^\s*{")
+    if p.match(alist_raw):
+        print(f"{basename(allowlist_path)} is JSON-formatted; converting to IMA policy")
+        alist_json = json.loads(alist_raw)
+        ima_policy = copy.deepcopy(ima.EMPTY_IMA_POLICY)
+        ima_policy["meta"]["timestamp"] = str(datetime.datetime.now())
+        ima_policy["meta"]["generator"] = "convert-ima-policy"
+        for key in alist_json.keys():
```

Comment:
I would do this the other way around: loop through the keys from the empty policy we know is good, rather than whatever keys might be present in the user data. That way if there are missing keys or bad keys they aren't just ignored or copied over, but we can give actual error messages. You'll still need your special handling for hashes vs digests.

Reply:
@mpeters Done in both loops of this type in the script, and I also added more debug info in case keys aren't found.